jgrapht-core/src/test/java/org/jgrapht/alg/tour/PalmerHamiltonianCycleTest.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.tour;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+import org.jgrapht.GraphTests;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.SimpleGraph;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+
+import static org.jgrapht.alg.tour.TwoApproxMetricTSPTest.assertHamiltonian;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class PalmerHamiltonianCycleTest {
+
+    /**
+     * Small graph of 4 nodes.
+     */
+    @Test
+    public void testSmallGraph(){
+        Graph<String, DefaultEdge> graph = new SimpleGraph<>(DefaultEdge.class);
+
+        graph.addVertex("A");
+        graph.addVertex("B");
+        graph.addVertex("C");
+        graph.addVertex("D");
+
+        graph.addEdge("A", "B");
+        graph.addEdge("A", "C");
+
+        graph.addEdge("B", "D");
+        graph.addEdge("C", "D");
+
+        GraphPath<String, DefaultEdge> tour = new PalmerHamiltonianCycle<String, DefaultEdge>().getTour(graph);
+
+        assertNotNull(tour);
+        assertHamiltonian(graph, tour);
+    }
+
+    /**
+     * Test that contains a simple cycle of 10 nodes.
+     * The graph has a Hamiltonian cycle but it doesn't meet Ore's condition.
+     */
+    @Test (expected = IllegalArgumentException.class)
+    public void testLineGraph(){
+        Graph<Integer, DefaultEdge> graph = new SimpleGraph<>(DefaultEdge.class);
+
+        for (int i = 0; i < 10; i++){
+            graph.addVertex(i);
+        }
+
+        for (int i = 0; i < 10; i++){
+            graph.addEdge(i, (i + 1) % 10);
+        }
+
+        GraphPath<Integer, DefaultEdge> tour = new PalmerHamiltonianCycle<Integer, DefaultEdge>().getTour(graph);
+
+        assertNotNull(tour);
+        assertHamiltonian(graph, tour);
+    }
+
+    /**
+     * Test with 500 randomly generated graphs.
+     * Method of generation: randomly add edges while the graph doesn't have Ore's property
+     */
+    @Test
+    public void testRandomGraphs(){
+        Random random = new Random(0xDEAD);
```

Comment:
Consider adding one more similar test which does not fix the random generator's seed. We usually do this, in order to catch more errors.

Reply:
I have added two more tests (one of each) with random seeds.